pgjdbc/src/main/java/org/postgresql/core/v3/ConnectionFactoryImpl.java

```
@@ -740,6 +767,51 @@ private static void doAuthentication(PGStream pgStream, String host, String user
     boolean saslHandshakeCompleted = false;
     ChannelBinding channelBinding = ChannelBinding.of(info);
 
+    // Parse requireAuth property for authentication method validation
+    String requireAuth = PGProperty.REQUIRE_AUTH.getOrDefault(info);
+    EnumSet<AuthMethod> allowedMethods = EnumSet.noneOf(AuthMethod.class);
+    EnumSet<AuthMethod> rejectedMethods = EnumSet.noneOf(AuthMethod.class);
+    if (requireAuth != null) {
+      boolean hasPositive = false;
+      boolean hasNegative = false;
+      EnumSet<AuthMethod> seenMethods = EnumSet.noneOf(AuthMethod.class);
+      
+      for (String method : requireAuth.split(",")) {
+        method = method.trim();
+        if (method.startsWith("!")) {
+          hasNegative = true;
+        } else {
+          hasPositive = true;
+        }
+      }
+      
+      if (hasPositive && hasNegative) {
+        throw new PSQLException(GT.tr("requireAuth cannot mix positive and negative authentication methods"), PSQLState.INVALID_PARAMETER_VALUE);
+      }
+      
+      for (String method : requireAuth.split(",")) {
+        method = method.trim();
+        AuthMethod authMethod;
+        if (method.startsWith("!")) {
+          authMethod = AuthMethod.fromString(method.substring(1));
+          if (authMethod != null) {
```

We're silently skipping unknown methods here. Is that intentional?

---

(Because the `fromString(...)` is going to return null)